[Lim Ai Lin] iP

LimAiLin opened the pull request:
# Dino
---
Dino :sauropod: is a task manager that helps you manage your tasks.
> "RAWR!" - Dino
---
It is
- easy to learn
- fast
and it is **FREE**!
---
## Usage:
1. Download it from [here](https://github.com/LimAiLin/ip/releases/tag/A-Jar)
2. Navigate to the jar file in your terminal
3. run `java -jar duke.jar`
---
## Features:
- [ ] Manage To-Dos
- [ ] Manage Deadlines
- [ ] Manage Events

### Example:
```sh
todo cs2100 lab
Dino:
	added: [T][ ] cs2100 lab
	You have 1 task!

deadline op1 /by 20/9/2022
Dino:
	added: [D][ ] op1 (by: Sep 20 2022)
	You have 2 tasks!

list
Dino:
	1. [T][ ] cs2100 lab
	2. [D][ ] op1 (by: Sep 20 2022)
```

Commit:
Add Gradle support

LimAiLin committed:
Level-1

LimAiLin committed:
Level-2

LimAiLin committed:
Level-3

LimAiLin committed:
Level-3 with A-Classes

LimAiLin committed:
Level-4

LimAiLin committed:
A-TextUiTesting

LimAiLin committed:
Level-5

LimAiLin committed:
Level-5

LimAiLin committed:
Level-6

LimAiLin committed:
A-Enums

LimAiLin committed:
A-Enums

LimAiLin committed:
A-Enums

LimAiLin committed:
Level-7

LimAiLin committed:
Level-7

LimAiLin committed:
Level-8

LimAiLin committed:
Merge branch 'branch-Level-8'

LimAiLin committed:
Merged

LimAiLin committed:
A-MoreOOP

LimAiLin committed:
A-Packages

Ssatu commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,73 @@
+import duke.task.Task;
+import duke.task.ToDo;
+
+public class AddCommand extends Command {
```
Hi, just a thought on naming convention. 
Might this be slightly confusing as a class name due to the use of a verb at the front? After some looking I can see that the class here refers to a command that involves adding of tasks. 

_Perhaps it could even be renamed to CommandToAdd!_

Ssatu commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,41 @@
+
+    public String getDescription() { return description; }
+
+    public static int lsSize() { return num; }
```
I'm not sure if I'm right but I believe this method serves to return the list size? If so, I think it would be a plus to take advantage of the naming convention as you have for other methods - i.e. getSize() 😄

jq1836 replied:
Similar to Ssatu's comment! Try to use clearer names for your methods!

Ssatu reviewed (commented):
Really clean code!
Left some comments on naming, but overall great job - I especially liked your use of enums when parsing input.

jq1836 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,46 @@
+
+import duke.command.Command;
+
+import java.io.*;
```
Try not to make use of wildcard imports! They are inefficient and are not as explicit as importing the specific class.

jq1836 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,50 @@
+        DELETE
+    }
+
+    public Command parse(String command) throws DukeException {
```
Very nice, clean and succinct parse method. Great job!

jq1836 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,35 @@
+        this.tasks = tasks;
+    }
+
+    public void list() {
```
Might be clearer to name this method printList.

jq1836 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,63 @@
+        System.out.println("\tOh no! You have more things to complete:\n\t" + myTask);
+    }
+
+    public void newLine() {
```
Perhaps a better method name would be printNewLine.

jq1836 reviewed (commented):
Generally decent code quality and great organization of packages! Some parts could be better abstracted to avoid boilerplate (repeated) code. Also, I understand that you haven't got around to writing up your Java Docs. Try to do so ASAP, it makes your code much more readable and understandable. I love your dino's personality too! :) Overall great attempt!

LimAiLin committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

LimAiLin committed:
A-JUnit

LimAiLin committed:
A-JavaDoc

LimAiLin committed:
A-CodingStandard

LimAiLin committed:
Level-9

LimAiLin committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Ui.java

ReubenChay commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,56 @@
+
+import duke.command.Command;
+
+import java.io.*;
```
similarly, wildcard imports are against 2103 coding standards!

ReubenChay commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,56 @@
+ */
+public class Duke {
+
+    private final Storage STORAGE;
```
minor issue but i think these dont need to be in all caps

ReubenChay commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,136 @@
+     * Creates a new Ui object.
+     */
+    public Ui() {
+        System.out.println("               __\n" +
```
cool dino

ReubenChay commented on src/main/java/duke/command/MarkCommand.java:
```diff
@@ -0,0 +1,63 @@
+ */
+public class MarkCommand extends Command {
+    private final String[] STR;
+    private final boolean MARK;
```
minor code standard just rename to ismarked or something!

ReubenChay reviewed (commented):
Overall really good abstraction. Code was concise and easy to read. Good job!

LimAiLin committed:
A-Checkstyle

rexong commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,120 @@
+        ArrayList<Task> ls = new ArrayList<>(100);
+        try {
+            if (this.FILE.createNewFile()) {
+                System.out.println("Dino created a new file: " + FILE.getName() + "\n");
+            } else {
+                BufferedReader br = new BufferedReader(new FileReader(FILE));
+                try {
+                    String current = br.readLine();
+                    while (current != null) {
+                        String[] str = current.split("\\|", 3);
+                        Commands myTask = Commands.valueOf(str[0].toUpperCase(Locale.ROOT));
+                        switch (myTask) {
+                        case DEADLINE:
+                            String[] dl = str[2].split("\\|", 2);
+                            Task deadline = new Deadline(dl[0], dl[1]);
+                            ls.add(deadline);
+                            if (Objects.equals(str[1], "1")) {
+                                deadline.markAsDone();
+                            }
+                            break;
+                        case TODO:
+                            Task todo = new ToDo(str[2]);
+                            ls.add(todo);
+                            if (Objects.equals(str[1], "1")) {
+                                todo.markAsDone();
+                            }
+                            break;
+                        case EVENT:
+                            String[] evnt = str[2].split("\\|", 2);
+                            Task event = new Event(evnt[0], evnt[1]);
+                            ls.add(event);
+                            if (Objects.equals(str[1], "1")) {
+                                event.markAsDone();
+                            }
+                            break;
+                        default:
+                            Ui.invalidTask();
+                        }
+                        current = br.readLine();
+                    }
```
I think here is a little hard to understand. Maybe can consider the use of guard clause.

rexong reviewed (commented):
Just added one comment.

LimAiLin committed:
Level-10

LimAiLin committed:
A-Assertions

LimAiLin committed:
A-CodeQuality

LimAiLin committed:
Merge pull request #2 from LimAiLin/branch-A-CodeQuality

A-CodeQuality

LimAiLin committed:
A-Assertions

LimAiLin committed:
Merge pull request #3 from LimAiLin/branch-A-Assertions

A-Assertions

LimAiLin committed:
Find command: make matching case insensitive

Find command is case sensitive.

A case insensitive find is more user-friendly because users cannot be
expected to remember the exact case of the keywords.

LimAiLin committed:
Merge remote-tracking branch 'origin/master'

LimAiLin committed:
D-Notes extension

New command for user to add notes, notes are different from other tasks because they don't need to be marked as done.

Support managing info about small snippets of textual information the user wants to record e.g., one's own waist size, a name of a movie that the user wants to remember

To improve:
* Be able to delete notes
* Sort notes alphabetically

LimAiLin committed:
Merge pull request #4 from LimAiLin/branch-BCD

D-Notes extension

LimAiLin committed:
A-BetterGui

LimAiLin committed:
A-Personality

LimAiLin committed:
A-UserGuide

LimAiLin committed:
A-CodeQuality

LimAiLin committed:
Update README.md

LimAiLin committed:
Optimised the Chatbot:

* The find command can now find for notes with the matching keyword
* Updated javadocs

LimAiLin committed:
Updated javadocs

LimAiLin committed:
A-MoreErrorHandling

LimAiLin committed:
Fixed errors: incorrect exception thrown
